### ISSUE TITLE
Some fonts do not print on macOS 10

### ISSUE THREAD

**Issue by inferiorhumanorgans** [typst/typst#4708]
### Description

If I pick a font outside the standard 14, the resulting PDF doesn't "print" properly with macOS 10's Preview.app

```
#set page("us-letter", margin: (
  top: 6.24cm,
  bottom: 3.17cm,
  left: 6.71cm,
  right: 2.12cm,
))

//#let sans = "Helvetica Neue"
#let sans = "Zwo Ot"

#set text(font: sans, weight: "light", size: 9pt)
Hello, world.
```

I'm running typst 0.11.1 on macOS 14.  The PDF generated from the above will render properly on the screen and within the print dialog box on macOS 10.  However any sort of printing will result in no text being visible.  Neither printing to PDF (or "open in preview") nor actually printing to a printer will render any text.  If I switch the font to Helvetica Neue things work as expected.  I also see the same incorrect behavior with other fonts e.g. Cst Berlin West, LTC Californian Pro, and IBM Plex Sans (the latter is open source).

The PDF appears to render and "print" properly regardless of font on macOS 14, however I don't actually have a printer hooked up to that machine.

Note that the font, Zwo OT Light, is installed on both machines so this is perhaps not a font embedding issue.

### Operating system

macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by inferiorhumanorgans** [comment]
I looked a bit further and the issue seems to be TrueType vs CID 0C.  Both display properly in the older Preview.app, but only the TrueType fonts show up in print output.

```
$ pdffonts output/thomas.pdf
name                                 type              encoding         emb sub uni object ID
------------------------------------ ----------------- ---------------- --- --- --- ---------
BKPAYY+FiraCode-Light                CID TrueType      Identity-H       yes yes yes      6  0
HXQTWN+FiraCode-Regular              CID TrueType      Identity-H       yes yes yes     11  0
RZWTEY+MetaPro-Light-Identity-H      CID Type 0C       Identity-H       yes yes yes     16  0
```

Fira Code shows up in the print output, Meta Pro does not.

**Comment by LaurenzV** [comment]
Do you happen to have the ability to compile Typst from source from `main`? There were known printer issues before with CFF fonts, but the font subsetting code has been [rewritten since then](https://github.com/typst/typst/pull/4373), and it _should_ help with preventing issues with CFF fonts when printing. So in case you have the ability to test that, that would be great.

**Comment by inferiorhumanorgans** [comment]
[test-print.pdf](https://github.com/user-attachments/files/16558583/test-print.pdf)
[test.pdf](https://github.com/user-attachments/files/16558597/test.pdf)

Yeah, I'll take a look.  In the meantime I've attached the file that typst generated (test.pdf) and what Preview.app generated (test-print.pdf).

**Comment by inferiorhumanorgans** [comment]
This appears fixed as of 831062a.

**Comment by LaurenzV** [comment]
Glad to hear. :)
